lldb/include/lldb/Core/ValueObject.h

```diff
@@ -668,6 +699,32 @@ class ValueObject {
   CreateValueObjectFromData(llvm::StringRef name, const DataExtractor &data,
                             const ExecutionContext &exe_ctx, CompilerType type);
 
+  static lldb::ValueObjectSP
```

Comment:
Given targets can be big or little-endian, I'm a little worried about a "const void *bytes" that says nothing about what is in the bytes.  This is what DataBuffer's are for, why couldn't you use them?

Reply:
I agree  on this, but why can't we just use the above `CreateValueObjectFromData(llvm::StringRef name, const DataExtractor &data, const ExecutionContext &exe_ctx, CompilerType type);`? The `DataExtractor` in the mentioned existing API can own or share a pointer to data. Would the "const void *bytes" always be copied into the internal data in the `ValueObject` in this case?

Reply:
CreateValueObjectFromBytes was being called from CreateValueObjectfrom{APInt,Pointer,Bool,Nullptr}, receivng the pointer to the appropriate values in the input parameter, putting the bytes into a DataExtractor, then calling CreateValueObjectFromData with the DataExtractor. I thought that would handle things like correct endianness, etc.  However, I will move the "putting the bytes into the DataExtractor" into the callers for CreateValueObjectFromBytes & call CreateValueObjectFromData directly from each of them. This leads to a bit of code duplication, but I can see that it's safer. (This eliminates the CreateValueObjectFromBytes functions altogether).